Software-Construction/Factory-And-Abstract-Factory/Graphics/Screen.java

```diff
@@ -0,0 +1,177 @@
+package Graphics;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import Graphics.Shapes.ShapeType;
+
+/**
+ * @author Niraj Gupta
+ * Class to represent the screen on which the shapes are drawn.
+ */
+public class Screen{
+	private List<Shapes> listOfShapes = new ArrayList<>();
+	private final double XMAX, YMAX;
+
+	/**
+	 * Parameterized Constructor
+	 * @param xMax : The maximum length of the screen.
+	 * @param yMax : The maximum breadth of the screen.
+	 */
+	public Screen(double xMax, double yMax){
+		XMAX = xMax;
+		YMAX = yMax;
+	}
+
+	/**
+	 * Method to add shape on the screen
+	 * @param shape : The shape which is to be added on the screen.
+	 */
+	public void addShape(Shapes shape){
+		if(shape == null){
+			throw new AssertionError("Null object");
+		}
+		Point origin = shape.getOrigin();
+		if(origin.getX() < 0 || origin.getX() > XMAX || origin.getY() < 0 || origin.getY() > YMAX){
+			throw new AssertionError("Cannot create shape outside the screen");
+		}
+		listOfShapes.add(shape);
+		shape.setTimestamp(new Date());
+	}
+
+	/**
+	 * Method to remove the shape from the screen.
+	 * @param shape : The shape which is to be removed from the screen.
+	 */
+	public void deleteShape(Shapes shape){
+		if(listOfShapes.contains(shape)){
+			listOfShapes.remove(shape);
+		}else{
+			throw new AssertionError("The shape is not present on screen!");
+		}
+	}
+
+	/**
+	 * Method to remove all the shapes from the screen belonging to a particular screen type.
+	 * @param shapeType : The type of the shape which is to be removed.
+	 */
+	public void deleteShapeType(ShapeType shapeType){

```

Comment:
what if I wish to support a new shape type I mean I want to add a new shape of say moon type. How would you support that?